### PR TITLE
TGUI item information

### PR DESCRIPTION
**Pull request by BobJoga**
## About The Pull Request

Adds a button on the UI to open a tgui item information menu, which tells you armor and damage information about the item in  your active hand.
image

This is obviously inspired by Eris, and it replaces the armor stats on examine more.

image
image

## Why It's Good For The Game

Convenient way to check stats without code diving.

## Changelog
:cl:
add: Added a TGUI item information menu verb for items. It an be acessed via a verb.
/:cl:

### CODE REVIEW

**Comment by Gandalf2k15** [comment]
Normally we let TGUI compile serverside eeee

**Comment by Tename** [comment]
Ability to build stuff with metal sheets seems to be broken, it opens two UIs and both spazz out and dont respond

**Comment by ReynTheLord** [comment]
NO. this opens up the info tab WHENEVER you interact with some items, including the pda right now.

Additionally, this ACTIVELY encourages powergaming or judging stats of gear, as usually you'd have to go to wiki or even the damn github to look for stats, but now you can just see it on the fly.

fuck

**Comment by Gandalf2k15** [comment]
Testmerging.

**Comment by BobJoga** [comment]
> NO. this opens up the info tab WHENEVER you interact with some items, including the pda right now.
> 
> Additionally, this ACTIVELY encourages powergaming or judging stats of gear, as usually you'd have to go to wiki or even the damn github to look for stats, but now you can just see it on the fly.
> 
> fuck

This was a bug, in regards to opening on any interaction. Fixed now.
To the additionally part: Just stupid. The code is open source, the info is always public, this only makes it easier to get without requiring boring code diving. Powergaming will be the same.

**Comment by Owai-Seek** [comment]
Honestly, if anything, seeing what each item gives armor or damagewise would result in balance changes for things that are too broken, instead of the info for people "in the know."

Many people don't give a shit about armor at all unless they're playing security or antagonist, and those that do, outside security or emergencies, are probably powergaming regardless. A random person knowing a crowbar is less effective than a toolbox or pickaxe is fine.

**Comment by Weblure** [comment]
TIL wire cutters deal more damage than any other basic tools.
10/10 I approve of this PR

**Comment by Owai-Seek** [comment]
I'm noticing some items (such as Oxygen Tanks, and Floor Tiles) are opening a second menu.

File: modular_skyrat/code/game/objects/items.dm [hunk 1]
```diff
@@ -12,3 +16,39 @@
 		user.visible_message("<span class='danger'>[t]</span>")
 	if(grabsound)
 		playsound(src,grabsound,50,1)
+
+//TGUI item info menu
+/obj/item/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, \
```
Comment (TriggeredBoi) [inline modular_skyrat/code/game/objects/items.dm:21]: /obj/item owning this UI is very bad due to inheritance stuff, any other /obj/item that has a TGUI part will fuck up, such as oxygen tanks. You'll have to put the UI under a datum and set ui_host to the item. Not hard.

Comment (BobJoga) [reply modular_skyrat/code/game/objects/items.dm:21]: going to make an element

**Review by Gandalf2k15 — APPROVED** [review]

**Comment by Weblure** [comment]
Uh oh, stinky
image

**Comment by Arturlang** [comment]
You made it old tgui, not tgui 3.0 and all the TGUI UI names are uppercase for both words

**Pull request merged**